microservices/restful-api-design-guidelines.md

```diff
@@ -0,0 +1,45 @@
+# RESTful API Design Guidelines
+
+## Introduction
+This page documents some guidelines to be used when designing and specifying RESTful APIs for microservices within the Survey Data Collection (SDC) platform.
+
+## HATEOAS
+We will not be building full [HATEOAS](https://en.wikipedia.org/wiki/HATEOAS)-complaint APIs as there is currently no clear requirement for the SDC platform that justifies the additional development complexity and effort.
+
+## URIs
+URIs should be all lowercase. The primary means for identifying a resource to access is via its UUID, as per the object identity strategy. The UUID should appear after the plural resource name without further qualification:
```

Comment:
I agree with the lowercasing of the uris, but not sure about going into the detail of the uuid bit - that's quite RAS/RM specific at the moment?

Reply:
It is. I'd like to retain that detail, but I'll reword to make it clear the UUID stuff is as currently implemented by RAS/RM.